File: src/main/java/com/tang/intellij/lua/psi/impl/LuaPsiElementImpl.java
package com.tang.intellij.lua.psi.impl;

import com.intellij.extapi.psi.ASTWrapperPsiElement;
import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiReference;
import com.intellij.psi.PsiReferenceService;
import com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistry;
import com.tang.intellij.lua.psi.LuaPsiElement;
import org.jetbrains.annotations.NotNull;

/**
 * LuaPsiElement 基类
 * Created by TangZX on 2016/11/24.
 */
public class LuaPsiElementImpl extends ASTWrapperPsiElement implements LuaPsiElement {
    LuaPsiElementImpl(@NotNull ASTNode node) {
        super(node);
    }

    @NotNull
    @Override
    public PsiReference[] getReferences() {
        return ReferenceProvidersRegistry.getReferencesFromProviders(this, PsiReferenceService.Hints.NO_HINTS);
    }
}

File: src/main/java/com/tang/intellij/lua/psi/impl/LuaPsiElementImpl.kt
/*
 * Copyright (c) 2017. tangzx(love.tangzx@qq.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.tang.intellij.lua.psi.impl

import com.intellij.extapi.psi.ASTWrapperPsiElement
import com.intellij.lang.ASTNode
import com.intellij.psi.PsiReference
import com.intellij.psi.PsiReferenceService
import com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistry
import com.tang.intellij.lua.psi.LuaPsiElement

/**
 * LuaPsiElement 基类
 * Created by TangZX on 2016/11/24.
 */
open class LuaPsiElementImpl internal constructor(node: ASTNode) : ASTWrapperPsiElement(node), LuaPsiElement {

    override fun getReferences(): Array<PsiReference> {
        return ReferenceProvidersRegistry.getReferencesFromProviders(this, PsiReferenceService.Hints.NO_HINTS)
    }

    override fun getReference(): PsiReference? {
        return references.firstOrNull()
    }
}
